pkg/config/runtime.go

```diff
@@ -24,41 +24,41 @@ import (
 )
 
 type (
-	// Runtime represents the runtime view of the config.
+	// runtime represents the runtime view of the config.
 	// It is pre-validated and immutable.
 	// It can be safely used concurrently.
-	Runtime struct {
+	runtime struct {
 		Validated
 		// used to evaluate selectors
 		eval expr.PredicateEvaluator
 	}
 
-	// AspectSet is a set of aspects by name.
-	AspectSet map[string]bool
+	// KindSet is a set of aspects by kind.
+	KindSet int
```

Comment:
Can you define `SetKind(k kind)` and `AddKind(k kind)` methods on `KindSet` for `kind &= 1 << uint(k)` and `kind |= 1 << uint(k)` respectively? I think that'll be far more readable than the bit shifts littered around and ensures the code using `KindSet` doesn't have to blindly assume it's an `int`, or that `Kind` is an alias for `uint`.